tools/testing/upload_artifacts.py

```diff
@@ -140,3 +143,59 @@ def trigger_upload_test_stats_intermediate_workflow() -> None:
         },
     )
     print(x.text)
+
+
+def parse_xml_and_upload_json() -> None:
+    """
+    Parse xml test reports that do not yet have a corresponding json report
+    uploaded to s3, and upload the json reports to s3.
+    """
+    json_files = [
+        Path(f).with_suffix("")
+        for f in glob.glob(f"{REPO_ROOT}/test/test-reports/**/*.json", recursive=True)
```

Comment:
is this doing the recursive glob scan for every finished test? won't it be too slow?